### PR TITLE
TST: Add tests for to_file/read_file for spatialite format

### PR DESCRIPTION
**Pull request by theroggy**
Because writing to spatialite is now added as an example in to_file, it was mentioned in #2821 that it would be better to add a test on it as well...

### CODE REVIEW

**Comment by theroggy** [comment]
While writing the test I stumbled upon a specific case that seems to be giving trouble. When writing a PointZ geometry to a spatialite file using pyogrio this gives an error. Apparently SpatiaLite is pickyer than GeoPackage and doesn't want to save 3D points in a "Point"column. So it was necessary to explicitly specify geometry_type = "Point Z" to get this case working.

Not sure how to deal with it, because i pyogrio this is detemined using gdf.geometry.type, which returns "Point" for a 3D point... Not sure if this is by design.

**Comment by martinfleis** [comment]
The geom type of 3D points is probably also related to  #2824.

**Comment by theroggy** [comment]
Apparently the SQLite driver was only added to fiona in version 1.8.20 (2021-05-31) and the "minimal" version is now set to 1.8.18 (2020-11-17). Not sure how to deal with that. Add a skip to the test or ?

**Comment by martinfleis** [comment]
Yeah, I would skip the test. We'll be bumping the minimal to the supported one in two months anyway.

**Comment by theroggy** [comment]
> 

I added the skip...

File: geopandas/io/tests/test_file_geom_types_drivers.py [hunk 1]
```diff
@@ -261,15 +261,42 @@ def engine(request):
 
 def test_to_file_roundtrip(tmpdir, geodataframe, ogr_driver, engine):
     output_file = os.path.join(str(tmpdir), "output_file")
+    write_kwargs = {}
+    if ogr_driver == "SQLite":
+        write_kwargs["spatialite"] = True
+
+        # This if statment can be removed once minimal fiona version >= 1.8.20
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file_geom_types_drivers.py:268]:
```suggestion
        # This if statement can be removed once minimal fiona version >= 1.8.20
```

File: geopandas/io/tests/test_file_geom_types_drivers.py [hunk 2]
```diff
@@ -261,15 +261,42 @@ def engine(request):
...
+        if engine == "fiona":
+            import fiona
+            from packaging.version import Version
+
+            if Version(fiona.__version__) < Version("1.8.20"):
+                pytest.skip("SQLite driver only available from version 1.8.20")
+
+        # If only 3D Points, geometry_type needs to be specified for spatialite at the
+        # monent. This if can be removed once the following PR is released:
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file_geom_types_drivers.py:277]:
```suggestion
        # moment. This if can be removed once the following PR is released:
```

File: geopandas/io/tests/test_file_geom_types_drivers.py [hunk 3]
```diff
@@ -261,15 +261,42 @@ def engine(request):
...
+        # https://github.com/geopandas/pyogrio/pull/223
+        if (
+            engine == "pyogrio"
+            and len(geodataframe == 2)
+            and geodataframe.geometry[0] is None
+            and geodataframe.geometry[1] is not None
+            and len(geodataframe.geometry[1].coords[0]) > 2
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_file_geom_types_drivers.py:284]:
```suggestion
            and geodataframe.geometry[1].has_z
```

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

**Pull request merged**